internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go

```diff
@@ -854,6 +861,46 @@ func (c *ClusterManager) buildObservabilityCloudwatchLoggingCredentialsSecret()
 	}
 }
 
+func (c *ClusterManager) buildObservabilityEnterpriseCloudwatchLoggingCredentialsSecret(orgId string, clusterId string) *k8sCoreV1.Secret {
+	idx, enterpriseCredential := arrays.FindFirst(c.ObservabilityConfiguration.ObservabilityCloudWatchLoggingConfig.EnterpriseCredentials, func(enterpriseCredential observatorium.ObservabilityEnterpriseCloudwatchLoggingConfigCredentials) bool {
+		return orgId == enterpriseCredential.OrgId
+	})
+	glog.Info("IDX: %q", idx)
+	if idx != arrays.ElementNotFound {
```

Comment:
I think this can be refactored in the following way:

1. build the K8s secret with all the elements but StringData and store it in a variable
2. if the element is not found log the warning and return that built secret
3. if the element is found then build the stringDataMap and set the StringData field to the  K8s secret variable. Finally return the variable

Reply:
:+1: